composer/trainer/trainer.py

```diff
@@ -973,8 +973,8 @@ def __init__(
         self.state.evaluators = evaluators
 
         # Some algorithms require specific settings
-        self._backwards_create_graph = any(map(lambda x: x.backwards_create_graph, ensure_tuple(algorithms)))
-        self._find_unused_parameters = any(map(lambda x: x.find_unused_parameters, ensure_tuple(algorithms)))
+        self._backwards_create_graph = any(map(lambda x: x.backwards_create_graph, ensure_tuple(self.state.algorithms)))
+        self._find_unused_parameters = any(map(lambda x: x.find_unused_parameters, ensure_tuple(self.state.algorithms)))
```

Comment:
ensure_tuple is not needed. self.state__init__ takes care of it.

Reply:
Good catch!